File: dotnet/src/Model/Player/Buildings/PlayerBuildings.cs
using System;
using System.Collections.Generic;

namespace PlanetCS1 {
    class PlayerBuildings {
        public List<ITile> units {get;set;}
        int counter;
        public PlayerBuildings() {
            this.units = new List<ITile>();
            this.counter = 0;
        }

        List<int> getNextFactory() {
            if (this.units.Count == 0) {
                return new List<int>(){-1, -1};
            }
            else if (this.units.Count == 1 && this.units[0] is PFactory) {
                return new List<int>(){this.units[0].loc.row, this.units[0].loc.col};
            }
            int stop = this.units.Count;
            for (int i = 0; i < this.units.Count; ++i) {
                counter+= 1;
                if (counter == this.units.Count) {
                    counter = 0;
                }
                if (this.units[counter] is PFactory) {
                    return new List<int>(){this.units[counter].loc.row, units[counter].loc.col};
                }
            }
            return new List<int>(){-1, -1};
        }
    }
}

File: kotlinjs/old/PlayerBuildings.kt


class PlayerBuildings() {
    val units = mutableListOf<ITile>()
    private var counter = 0


    fun getNextFactory() : MutableList<Int> {
        if (units.size == 0) {
            return mutableListOf<Int>(-1, -1)
        }
        else if (units.size == 1 && units[0] is PFactory) {
            return mutableListOf<Int>(units[0].loc.row, units[0].loc.col)
        }
        val stop  = units.size
        for (i in 0 until stop) {
            counter += 1
            if (counter == units.size) {
                counter = 0
            }
            if (units[counter] is PFactory) {
                return mutableListOf<Int>(units[counter].loc.row, units[counter].loc.col)
            }
        }
        return mutableListOf<Int>(-1, -1)
    }
}